### ISSUE TITLE
Incorrect suggestion for SIM208

### ISSUE THREAD

**Issue by dosisod** [charliermarsh/ruff#3350]
The `SIM208` rule should be updated to only emit errors for `not not x` in boolean contexts, since changing `not not x` to just `x` can cause different results. When used in a non-boolean context, `bool(x)` should be used instead to ensure it results in a boolean type.

## Example

```python
x = 123
y = not not x
```

In the above example, `y` is `True`, but changing it to `y = x` make it `123` instead.

## Solution

Only suggest `x` when in a boolean context, and `bool(x)` everywhere else:

```python
y = not not x  # bool(x)

if not not x:  # x
    ...
elif not not x:  # x
    ...
elif f(not not x):  # f(bool(x))
    ...
```

I opened a similar issue on the [flake8-simplify](https://github.com/MartinThoma/flake8-simplify/issues/170) repo as well.

**Comment by charliermarsh** [comment]
Oh interesting, thanks, I think we do this already (or something like it) for `SIM210`.

**Comment by SiddharthDoley** [comment]
x=!!x 
Y=123

**Comment by crimsonknave** [comment]
I'm getting flagged for SIM208 in the exact example listed in the description. It does not suggest I use `bool(x)` and instead suggests to use `x`.

Python version: 3.11.9
Ruff version: 0.5.6
